src/TxHistory/useOnScroll.ts

```diff
@@ -2,36 +2,39 @@ import {useRef} from 'react'
 import {NativeScrollEvent, NativeSyntheticEvent, ScrollViewProps} from 'react-native'
 
 const noop = () => undefined
+const buffer = 100
 
 type Callbacks = {
   onScrollUp?: ScrollViewProps['onScroll']
   onScrollDown?: ScrollViewProps['onScroll']
 }
 export const useOnScroll = ({onScrollUp = noop, onScrollDown = noop}: Callbacks) => {
-  const scrollOffset = useRef(0)
+  const previousScrollOffset = useRef(0)
 
   const onScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
     const {nativeEvent} = event
-    const topBounce = nativeEvent.contentOffset.y <= 0
+    const topBounce = nativeEvent.contentOffset.y < 0
     const bottomBounce =
-      nativeEvent.contentSize.height - nativeEvent.contentOffset.y - nativeEvent.layoutMeasurement.height <= 0
+      nativeEvent.contentSize.height - nativeEvent.contentOffset.y - nativeEvent.layoutMeasurement.height < 0
+    const topReached = nativeEvent.contentOffset.y === 0
+    const scrollOffset = nativeEvent.contentOffset.y
 
     if (topBounce) return // ignore top bounce
     if (bottomBounce) return // ignore bottom bounce
 
-    const scrollUp = nativeEvent.contentOffset.y < scrollOffset.current
-    const scrollDown = nativeEvent.contentOffset.y > scrollOffset.current
+    const scrollUp = scrollOffset < previousScrollOffset.current - buffer
+    const scrollDown = scrollOffset > previousScrollOffset.current + buffer
```

Comment:
I've tested this on an emulator, and the header does not collapse when I scroll slowly. Is this expected?

Reply:
not sure how to write this better,
slow scroll without buffer has jitters